kaldb/src/main/java/com/slack/kaldb/metadata/recovery/RecoveryTaskMetadata.java

```diff
@@ -11,12 +11,19 @@
   public final String partitionId;
   public final long startOffset;
   public final long endOffset;
+  public final long createdTimeUtc;
 
-  public RecoveryTaskMetadata(String name, String partitionId, long startOffset, long endOffset) {
+  public RecoveryTaskMetadata(
+      String name, String partitionId, long startOffset, long endOffset, long createdTimeUtc) {
     super(name);
     this.partitionId = partitionId;
     this.startOffset = startOffset;
     this.endOffset = endOffset;
+    this.createdTimeUtc = createdTimeUtc;
```

Comment:
Please add some arg checks here like partitionId not null and createdTimeUtc to be a positive number etc so we can catch any data issues?

Reply:
Done